vm-setup/roles/libvirt/tasks/network_setup_tasks.yml

```diff
@@ -22,7 +22,7 @@
     nodes: "{{ vm_nodes }}"
     networks: "{{ networks }}"
   register: node_mac_map
-  when: vm_nodes
+  when: not vm_nodes|bool
```

Comment:
Why `not`? Shouldn't this be `when: vm_nodes is defined`?

Reply:
when: vm_nodes is ok until ansible 2.12. Now it's just a [DEPRECATION WARNING]
https://github.com/ansible/ansible/issues/53428

Reply:
That's not what I'm asking, please see my comment.

Reply:
Right `vm_nodes` is a list https://github.com/metal3-io/metal3-dev-env/blob/master/vm-setup/roles/common/tasks/main.yml#L8

So this should instead be changed to check if the length of that list is non-zero AFAICS?

Reply:
Yep, something like `vm_nodes | length > 0`